drivers/console/semihost_console.c

```diff
@@ -13,16 +13,23 @@ extern void __stdout_hook_install(int (*fn)(int));
 
 static int semihost_console_out(int ch)
 {
-	static unsigned char c;
-
-	c = ch;
-	__asm__ __volatile__ (
-		"movs	r1, %0\n"
-		"movs	r0, #3\n"
-		"bkpt	0xab\n"
-		:
-		: "r" (&c)
-		: "r0", "r1");
+	static unsigned short n;
+	static unsigned char buf[CONFIG_SEMIHOST_CONSOLE_OUTPUT_BUFFER_SIZE];
```

Comment:
Not a fan of declaring statics in a function like this, since it obfuscates the code a bit I think?